docs/DeveloperGuide.md

```diff
@@ -6,17 +6,146 @@
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+Our program is structured as pictured in the following class diagram:
+![ClassDiagram.png](diagrams/ClassDiagram.png)
```

Comment:
Would be good to include the access modifier. It will be good that you spread out the class diagram as well so that the association is not overlapped.